central/sensor/service/connection/sensorevents.go

```diff
@@ -12,15 +13,21 @@ import (
 	"github.com/stackrox/rox/generated/internalapi/central"
 	"github.com/stackrox/rox/pkg/concurrency"
 	"github.com/stackrox/rox/pkg/reflectutils"
+	"github.com/stackrox/rox/pkg/sync"
 	"github.com/stackrox/rox/pkg/utils"
 )
 
 const workerQueueSize = 16
```

Comment:
Does this need to be doubled for the queue holding nodes and nodeInventories now?

Reply:
Could you please clarify what specifically you want to double for the worker queues? Also, I assume you are talking about the deduping queues for Node and NodeIventory if you could clarify? Need that information to be able to answer your question.

Reply:
Sure, here is my thinking. We used to have 2 separate queues for node and nodeInventories - each of capacity 16. Now, we place both kind of messages into a single queue of size 16, which means that the capacity of 16 needs to be shared between the nodes and nodeInventories, which (on average) reduces the capacity to 8 messages of one type.
Is this correct?

I do not know what are the consequences of having less capacity in the queue, so I would also like to understand what would be the consequence of setting it to, for example, 1.

Reply:
Oh, I see now. The `workerQueueSize` is the number of shards inside the main worker queue. Each worker queue has 16 shards; the shard key is the resource ID. That means each resource is handled serialized in the same Q, but multiple resources can be handled concurrently. The name `workerQueueSize` is just a lousy name IMO: it should be `shardCount` or similar.